README.md

```diff
@@ -0,0 +1,82 @@
+# bunyan-env-config
+
+A small configuration wrapper around Bunyan logging.  This module's main
+purpose is to contain all the taskcluster boiler plate and configuration logic
+around logging with Bunyan.
+
+We want to use Bunyan logging in our applications so that our logs are easy to
+parse and obtain insights from.  When we use bunyan, we can start putting lots
+of things in our logs easily and use those to monitor our systems.  We can also
+submit things to security monitoring systems a lot easier if we can combine the
+logging messages with the security forwarding.
+
+One of the complexities of using bunyan is that it's a very configurable library.
+Our usage is currently limited to printing to standard output and forwarding
+that output to a log aggregator.
+
+## Log Level Configuration
+write this when we finalise the format
+
+## Api
+The logging object returned by this module is just a bunyan object with a
+debugCompat property on the root logger.  All bunyan APIs after the initial
+configuration continue to apply.  When reading bunyan docs, this library
+replaced only the bunyan.createLogger call.
+
+## Usage
+The intention is for each application to only call the function exported by
+this module once.  In a single js-file library, just do that in your code
+directly.  In larger applications, the suggested method is to create a small
+file (`lib/log.js`) file which sets up the logging for your whole application.
+Since `require()` will cache that module, you'll be sharing the same root
+logger in all of your code.
+
+This library targets applications which run on systems like Heroku which
+capture stdio streams and forward them onto a log aggregator.
+
+Right now, we're relying on applications to figure out for themselves how
+they'd like to figure out what object to pass into logging function.  One idea
+would be to have a file like:
+```javascript
+module.exports = require('bunyan-env-config')('my-app', {});
```

Comment:
Okay, this now makes a little sense...

So `taskcluster-client` will keep using `debug`, it'll never use this logging method... Correct? And the same goes for all of our libraries. Unless there is special cases where we want libraries to actually send production logs for us.

I imagine that we don't want or ever need debug logs sent... which is why using `debug` in libraries makes sense, rather than passing in a `log` object from the application. Also `debug` is zero overhead if `DEBUG` env var isn't set, I'm not sure that true for this logger...

@jhford, is this correct, libraries shouldn't use this... Only applications and libraries that wish to send special production log messages. IMO we should run our applications with `DEBUG` env undefined once this rolls out. Agree?

This way `debug` module is just for debugging locally when developing a library. Not for logging. I think that's fair as `debug` is really efficient and easy to use no passing objects around.